File: 46.Permutations.cpp
class Solution {
    public:
    vector<vector<int>> permute(vector<int>& nums) {
        vector<vector<int>> res;
        vector<int> curr;
        permutationHelper(res, nums, curr, nums.size());
        return res;
    }
    
    void permutationHelper(vector<vector<int>>& res, vector<int>& nums, vector<int>& curr, int count ){
        if (count == 0){
            res.push_back(curr);
            return;
        }
        else {
            for (int i=0; i<nums.size(); i++){
                if (find(curr.begin(), curr.end(), nums[i]) == curr.end()){
                    // if (!count(curr.begin,curr.end(), nums[i])){
                    curr.push_back(nums[i]);
                    permutationHelper(res, nums, curr, count-1);
                    curr.pop_back();
                }
            }
        }
    }
};

File: 46.Permutations.py
class Solution:
    def permute(self, nums: List[int]) -> List[List[int]]:
        res = []
        self.helper(nums, [], res)
        return res
    
    def helper(self, nums, curr, res):
        if len(curr) == len(nums):
            res.append(curr)
            return
        for num in nums:
            if num in curr:
                continue
            else:
                self.helper(nums, curr+[num], res)
        return